validations/transactWriteItems.js

```diff
@@ -0,0 +1,194 @@
+              notNull: true,
+              children: 'AttrStruct<ValueStruct>',
+            },
+          },
+        },
+        Update: {
+          type: 'FieldStruct<Update>',
+          children: {
+            TableName: {
+              type: 'String',
+            },
+            ExpressionAttributeValues: {
+              type: 'Map<AttributeValue>',
+              children: 'AttrStruct<ValueStruct>',
+            },
+            ExpressionAttributeNames: {
+              type: 'Map<java.lang.String>',
+              children: 'String',
+            },
+            ReturnValuesOnConditionCheckFailure: {
+              type: 'String',
+              enum: [ 'ALL_OLD', 'NONE' ],
+            },
+            ConditionExpression: {
+              type: 'String',
+            },
+            Key: {
+              type: 'Map<AttributeValue>',
+              notNull: true,
+              children: 'AttrStruct<ValueStruct>',
+            },
+            UpdateExpression: {
+              type: 'String',
+            },
+          },
+        },
+        Delete: {
+          type: 'FieldStruct<Delete>',
+          children: {
+            TableName: {
+              type: 'String',
+            },
+            ExpressionAttributeValues: {
+              type: 'Map<AttributeValue>',
+              children: 'AttrStruct<ValueStruct>',
+            },
+            ExpressionAttributeNames: {
+              type: 'Map<java.lang.String>',
+              children: 'String',
+            },
+            ReturnValuesOnConditionCheckFailure: {
+              type: 'String',
+              enum: [ 'ALL_OLD', 'NONE' ],
+            },
+            ConditionExpression: {
+              type: 'String',
+            },
+            Key: {
+              type: 'Map<AttributeValue>',
+              notNull: true,
+              children: 'AttrStruct<ValueStruct>',
+            },
+          },
+        },
+        ConditionCheck: {
+          type: 'FieldStruct<ConditionCheck>',
+          children: {
+            TableName: {
+              type: 'String',
+            },
+            ExpressionAttributeValues: {
+              type: 'Map<AttributeValue>',
+              children: 'AttrStruct<ValueStruct>',
+            },
+            ExpressionAttributeNames: {
+              type: 'Map<java.lang.String>',
+              children: 'String',
+            },
+            ReturnValuesOnConditionCheckFailure: {
+              type: 'String',
+              enum: [ 'ALL_OLD', 'NONE' ],
+            },
+            ConditionExpression: {
+              type: 'String',
+            },
+            Key: {
+              type: 'Map<AttributeValue>',
+              notNull: true,
+              children: 'AttrStruct<ValueStruct>',
+            },
+          },
+        },
+      },
+    },
+  },
+}
+
+exports.custom = function (data, store) {
+  var i, request, msg, key
+  for (i = 0; i < data.TransactItems.length; i++) {
+    request = data.TransactItems[i]
```

Comment:
subject to style debate, or maybe performance?

I'd rather move the variable declaration into the loop:

```suggestion
  for (const request of data.TransactItems) {
    let msg;
```

Reply:
Agree